File: example/src/main/java/net/dontdrinkandroot/wicket/example/component/NavBarForm.java
package net.dontdrinkandroot.wicket.example.component;

import org.apache.wicket.markup.html.form.Form;
import org.apache.wicket.markup.html.panel.IMarkupSourcingStrategy;
import org.apache.wicket.markup.html.panel.PanelMarkupSourcingStrategy;
import org.apache.wicket.model.Model;

import net.dontdrinkandroot.wicket.bootstrap.behavior.NavBarFormBehavior;
import net.dontdrinkandroot.wicket.bootstrap.component.button.AjaxSubmitButton;
import net.dontdrinkandroot.wicket.bootstrap.component.form.formgroup.FormGroupTextField;


public class NavBarForm extends Form<Void>
{

	public NavBarForm(String id)
	{
		super(id);
	}

	@Override
	protected void onInitialize()
	{
		super.onInitialize();
		this.add(new NavBarFormBehavior());
		FormGroupTextField<String> searchGroup =
				new FormGroupTextField<String>("searchGroup", Model.of("Search"), new Model<String>());
		searchGroup.setLabelScreenReaderOnly(true);
		this.add(searchGroup);
		this.add(new AjaxSubmitButton("submit").setBody(Model.of("Search")));
	}

	@Override
	protected IMarkupSourcingStrategy newMarkupSourcingStrategy()
	{
		return new PanelMarkupSourcingStrategy(false);
	}

}

File: example/src/main/kotlin/net/dontdrinkandroot/wicket/example/component/NavbarForm.kt
package net.dontdrinkandroot.wicket.example.component

import net.dontdrinkandroot.wicket.bootstrap.behavior.form.inlineForm
import net.dontdrinkandroot.wicket.bootstrap.component.button.AjaxSubmitButton
import net.dontdrinkandroot.wicket.bootstrap.component.form.RepeatingForm
import net.dontdrinkandroot.wicket.bootstrap.component.form.formgroup.FormGroupInputText
import org.apache.wicket.markup.repeater.RepeatingView
import org.apache.wicket.model.Model

class NavbarForm(id: String) : RepeatingForm<Void>(id) {

    override fun onInitialize() {
        super.onInitialize()
        add(inlineForm())
    }

    override fun populateFormGroups(formGroupView: RepeatingView) {
        super.populateFormGroups(formGroupView)
        val searchGroup = FormGroupInputText(formGroupView.newChildId(), Model(), Model("Search"))
        searchGroup.setLabelScreenReaderOnly(true)
        formGroupView.add(searchGroup)
    }

    override fun populateActions(buttonView: RepeatingView)
    {
        buttonView.add(AjaxSubmitButton(buttonView.newChildId(), labelModel = Model("Search")))
    }
}